src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -18,6 +18,8 @@ public sealed class CSharpParseOptions : ParseOptions, IEquatable<CSharpParseOpt
         /// </summary>
         public static readonly CSharpParseOptions Default = new CSharpParseOptions();
 
+        private IReadOnlyDictionary<string, string> _features;
```

Comment:
I think we should make it `ImmutableDictionary` on the philosophy "be specific in what you provide and lenient in what you accept."

Of course, with the caveat that in perf optimization cases my rule changes to "whatever will make it faster."